src/PowerShellEditorServices/Debugging/DebugService.cs

```diff
@@ -122,62 +95,85 @@ public DebugService(PowerShellContext powerShellContext)
-                                    ast => (ast is BreakStatementAst || ast is ContinueStatementAst), true);
-
-                            // If this isn't advanced syntax then the conditions string should be a simple
-                            // expression that needs to be wrapped in a "if" test that conditionally executes
-                            // a break statement.
-                            if (breakOrContinueStatementAst == null)
-                            {
-                                string wrappedCondition = $"if ({breakpoint.Condition}) {{ break }}";
-                                actionScriptBlock = ScriptBlock.Create(wrappedCondition);
-                            }
-
-                            psCommand.AddParameter("Action", actionScriptBlock);
+                            resultBreakpointDetails.Add(breakpoint);
+                            continue;
                         }
-                        catch (ParseException ex)
+
+                        psCommand.AddParameter("Action", actionScriptBlock);
+                    }
+
+                    IEnumerable<Breakpoint> configuredBreakpoints =
+                        await this.powerShellContext.ExecuteCommand<Breakpoint>(psCommand);
+
+                    resultBreakpointDetails.AddRange(
+                        configuredBreakpoints.Select(BreakpointDetails.Create));
+                }
+            }
+
+            return resultBreakpointDetails.ToArray();
+        }
+
+        /// <summary>
+        /// Sets the list of line breakpoints for the current debugging session.
+        /// </summary>
+        /// <param name="scriptFile">The ScriptFile in which breakpoints will be set.</param>
+        /// <param name="breakpoints">BreakpointDetails for each breakpoint that will be set.</param>
+        /// <param name="clearExisting">If true, causes all existing breakpoints to be cleared before setting new ones.</param>
+        /// <returns>An awaitable Task that will provide details about the breakpoints that were set.</returns>
+        public async Task<FunctionBreakpointDetails[]> SetFunctionBreakpoints(
+            FunctionBreakpointDetails[] breakpoints,
+            bool clearExisting = true)
+        {
+            var resultBreakpointDetails = new List<FunctionBreakpointDetails>();
+
+            if (clearExisting)
+            {
+                await this.ClearCommandBreakpoints();
+            }
+
+            if (breakpoints.Length > 0)
+            {
+                foreach (FunctionBreakpointDetails breakpoint in breakpoints)
+                {
+                    PSCommand psCommand = new PSCommand();
+                    psCommand.AddCommand(@"Microsoft.PowerShell.Utility\Set-PSBreakpoint");
+                    psCommand.AddParameter("Command", breakpoint.Name);
+
+                    // Check if this is a "conditional" line breakpoint.
+                    if (breakpoint.Condition != null)
```

Comment:
Maybe I'm asking the wrong question, just confused by seeing this in the SetFunctionBreakpoints method.  Maybe the function name is wrong?